openforcefield/propertycalculator/protocols.py

```diff
@@ -0,0 +1,1053 @@
+#!/usr/bin/env python
+
+# =============================================================================================
+# MODULE DOCSTRING
+# =============================================================================================
+
+"""
+Protocol API.
+
+Authors
+-------
+* Simon Boothroyd <simon.boothroyd@choderalab.org>
+
+"""
+
+
+# =============================================================================================
+# GLOBAL IMPORTS
+# =============================================================================================
+
+import copy
+import logging
+
+import mdtraj
+
+import arch.bootstrap
+
+import numpy as np
+
+from os import path
+
+from enum import Enum
+
+from pymbar import timeseries
+
+from openeye import oechem, oeomega
+
+from openforcefield.utils import packmol
+from openforcefield.utils.exceptions import XmlNodeMissingException
+from openforcefield.typing.engines import smirnoff
+
+from simtk import openmm, unit
+from simtk.openmm import app
+
+
+# =============================================================================================
+# Protocols
+# =============================================================================================
+
+class ProtocolInputReference:
+    """Stores a reference to a required input from another protocol.
+
+    Each node represents a protocol to be executed.
+
+    Parameters
+    ----------
+    protocol_id: str
+        The identity of the other protocol.
+    property_name: str
+        The name of the property to inherit from the other protocol.
+    """
+
+    def __init__(self, input_property, protocol_id, property_name):
+
```

Comment:
Let's standardize and put __init__ argument docstrings here instead of in class definition